src/main/java/org/spongepowered/api/advancement/criteria/AdvancementCriterion.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.advancement.criteria;
+
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.advancement.Advancement;
+import org.spongepowered.api.advancement.criteria.trigger.Trigger;
+import org.spongepowered.api.util.ResettableBuilder;
+import org.spongepowered.api.util.generator.dummy.DummyObjectProvider;
+
+import java.util.Collection;
+import java.util.Optional;
+import java.util.Set;
+
+/**
+ * Represents a criterion that should be acquired
+ * to unlock a {@link Advancement}.
+ */
+public interface AdvancementCriterion {
+
+    /**
+     * Represents an empty criterion, this means that nothing has to be achieved to
+     * unlock a {@link Advancement}.
+     */
+    AdvancementCriterion EMPTY = DummyObjectProvider.createFor(AdvancementCriterion.class, "EMPTY");
+
+    /**
+     * Attempts to build a new AND operation with the given {@link AdvancementCriterion}s.
+     *
+     * @param criteria The other criteria
+     * @return The and operation
+     */
+    static AdvancementCriterion buildAnd(AdvancementCriterion... criteria) {
```

Comment:
You can make it `static AdvancementCriterion and(AdvancementCriterion adv1, AdvancementCriterion... therest)`